### PR TITLE
fix: master should exit when EADDRINUSE error

### PR DESCRIPTION
**Pull request by popomore**
##### Checklist

- [x] `npm test` passes
- [x] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [x] commit message follows commit guidelines

##### Description of change

app_worker will listen error event of server when starting,
if it emits app_worker will exit with code 1,
and master will exit with same code.

Closes https://github.com/eggjs/egg/issues/445

### CODE REVIEW

File: lib/master.js [hunk 1]
```diff
@@ -215,6 +215,7 @@ class Master extends EventEmitter {
 
     let debugPort = process.debugPort;
     cluster.on('fork', worker => {
+      worker.disableRefork = true;
```
Comment (popomore) [inline lib/master.js]: @fengmk2 未启动完成也不 refork

File: lib/app_worker.js [hunk 2]
```diff
@@ -46,6 +46,12 @@ function startServer() {
     server = require('http').createServer(app.callback());
   }
 
+  server.on('error', err => {
+    consoleLogger.error('[app_worker] server got error: %s, code: %s',
```
Comment (atian25) [inline lib/app_worker.js]: consoleLogger?

File: lib/master.js (hunk 1 above)
Comment (atian25) [reply lib/master.js]: 启动成功后，worker 死掉，新重启的 worker 呢？

Comment (popomore) [reply lib/master.js]: Worker 死了，master 就退出了

Comment (atian25) [reply lib/master.js]: 这种场景：全部 worker 都启动成功后，某次请求导致某个 worker 挂了，重新启动的新 worker。

`worker.disableRefork = !this.isAllAppWorkerStarted;`

File: lib/master.js [hunk 3]
```diff
@@ -420,6 +421,11 @@ class Master extends EventEmitter {
 
     this.isAllAppWorkerStarted = true;
 
+    for (const id in cluster.workers) {
```
Comment (atian25) [inline lib/master.js]: 新重启的 worker 进不来这段逻辑，上面已经 return 了

File: lib/master.js (hunk 1 above)
Comment (popomore) [reply lib/master.js]: 下面不是有设置

Comment (atian25) [reply lib/master.js]: 新加的那段循环？那段上面有一个 return

https://github.com/eggjs/egg-cluster/blob/a2bc2b2a1a89ba0e3920952977090e89163f6d57/lib/master.js#L418-L420

**Comment by popomore** [comment]
https://github.com/eggjs/egg-mock/pull/60

File: lib/master.js (hunk 1 above)
Comment (popomore) [reply lib/master.js]: 补充了一个逻辑，加上单测了

**Comment by popomore** [comment]
这个可以放放，等 @fengmk2 回来再看看

**Comment by popomore** [comment]
@fengmk2 看看

**Comment by fengmk2** [comment]
1.12.3

**Pull request merged**